packages/rspack/tests/cases/parsing/issue-5786/index.js

```diff
@@ -0,0 +1,15 @@
+class A {
+	constructor(
+		func = () => {
+			this.a = 1;
+		}
+	) {
+		func();
+	}
+}
+
+it("ensure top level is right in constructor parameters", () => {
+	const a = new A();
+	expect(a.a).toBe(1);
+	expect(typeof a).toBe("object");
+});
```

Comment:
The inclusion of ﻿`export default ''` is necessary as it allows this file to be recognized as `﻿js/esm` instead of `﻿js/auto`.